infrastructure/src/main/java/org/corfudb/infrastructure/SequencerServer.java

```diff
@@ -153,6 +153,10 @@ public SequencerServer(ServerContext serverContext) {
             globalLogTail.set(initialToken);
         }
 
+        if ((Boolean) opts.get("--single")) {
+            readyStateEpoch = serverContext.getNewSingleNodeLayout().getEpoch();
```

Comment:
How do you get the actual layout ?
You cannot bootstrap the node again as it will throw an Already bootstrapped exception.
If there is a reconfiguration change and the cluster moves to a new epoch, then the sequencer becomes NOT_READY as this epoch is now stale anyways. The management server then takes care of bootstrapping the new primary sequencer for the new epoch.

Reply:
installSingleNodeLayoutIfAbsent does not bootstrap the node. It installs a layout if it is not present. ManagementServer/LayoutServer constructors call the same function. Then just get the layout from the data store.

I'm not sure how an alreadybootstrap exception would be thrown, that isn't even thrown anywhere in ServerContext.

Reply:
So I'm just saying that once we start a server as a single node (-s) we cannot bootstrap the node again later. As you said, the LayoutServer installs the layout from the local data store.

Reply:
Done.